### DISCUSSION TITLE
Show the plugin native histogram in Dashboard view

### DISCUSSION THREAD

**Discussion by RoryCrispin** [grafana/clickhouse-datasource#403] [Ideas]
When adding a logs panel to a dashboard - the histogram view is not shown.
The only workaround seems to be creating a panel which generates a histogram as a bar chart.
It would be great to just use the native plugin histogram

**Comment by mshustov** [comment]
> When adding a logs panel to a dashboard - the histogram view is not shown.

I don't think it's supported out-of-the-box in dashboard view. See https://github.com/grafana/grafana/issues/63929

There is `logs volume` histogram in Explore view, though. It's supported in `clickhouse-datasource` starting from `v3.1.0`:

![2023-06-04_14-03-34](https://github.com/grafana/clickhouse-datasource/assets/3198181/5fabfb89-13e9-4369-bb5e-f0d1807b0bc4)

**Reply by RoryCrispin** [reply]
The explore view works great, but it's a shame that Grafana doesn't support this option in the dashboard view. It seems to be a difference in opinion about what a dashboard should be. We like to use dashboards as a tool with one logs panel connected to a set of variables for easy querying, while others see dashboards as an overview page of many panels, and the explore view should be used to 'dive deeper'.

**Comment by SpencerTorres** [comment]
This is indeed something baked in to the explore page, even in its implementation in the plugin code. You can see the query it runs in the network log. It's best to make it a separate panel, although we could make it easier to write this query by adding a switch to the logs query builder that changes the query to a histogram (similar to trace search vs trace id)
